pages/configuration-files/[...id].js

```diff
@@ -0,0 +1,219 @@
+// node_modules
+import PropTypes from "prop-types";
+import Link from "next/link";
+// components
+import AlternateAccessions from "../../components/alternate-accessions";
+import Attribution from "../../components/attribution";
+import Breadcrumbs from "../../components/breadcrumbs";
+import { FileDataItems } from "../../components/common-data-items";
+import {
+  DataArea,
+  DataAreaTitle,
+  DataPanel,
+  DataItemLabel,
+  DataItemValue,
+} from "../../components/data-area";
+import DerivedFromTable from "../../components/derived-from-table";
+import DocumentTable from "../../components/document-table";
+import { EditableItem } from "../../components/edit";
+import { FileHeaderDownload } from "../../components/file-download";
+import JsonDisplay from "../../components/json-display";
+import ObjectPageHeader from "../../components/object-page-header";
+import PagePreamble from "../../components/page-preamble";
+import SeparatedList from "../../components/separated-list";
+// lib
+import buildAttribution from "../../lib/attribution";
+import buildBreadcrumbs from "../../lib/breadcrumbs";
+import {
+  requestDocuments,
+  requestFileSets,
+  requestFiles,
+} from "../../lib/common-requests";
+import errorObjectToProps from "../../lib/errors";
+import FetchRequest from "../../lib/fetch-request";
+import {
+  checkForFileDownloadPath,
+  convertFileDownloadPathToFilePagePath,
+} from "../../lib/files";
+import { isJsonFormat } from "../../lib/query-utils";
+
+export default function ConfigurationFile({
+  attribution,
+  configurationFile,
+  fileSet = null,
+  seqSpecOf,
+  documents,
+  derivedFrom,
+  derivedFromFileSets,
+  fileFormatSpecifications,
+  isJson,
+}) {
+  return (
+    <>
+      <Breadcrumbs />
+      <EditableItem item={configurationFile}>
+        <PagePreamble>
+          <AlternateAccessions
+            alternateAccessions={configurationFile.alternate_accessions}
+          />
+        </PagePreamble>
+        <ObjectPageHeader item={configurationFile} isJsonFormat={isJson}>
+          <FileHeaderDownload file={configurationFile} />
+        </ObjectPageHeader>
+        <JsonDisplay item={configurationFile} isJsonFormat={isJson}>
+          <DataPanel>
+            <DataArea>
+              <FileDataItems
+                item={configurationFile}
+                fileSet={fileSet}
+              ></FileDataItems>
+            </DataArea>
+          </DataPanel>
+          {seqSpecOf?.length > 0 && (
```

Comment:
Not a big thing, but the `seqSpecOf` property is required (might be empty, which satisifies this requirement), so you can just do `seqSpecOf.length` without the question mark.